biggus/__init__.py

```diff
@@ -824,6 +832,47 @@ def masked_array(self):
         return array
 
 
+def _use_indexed_array(keys):
+    return ((isinstance(keys, tuple) and any(key is None for key in keys)) or
+            keys is None)
+
+
+class IndexedArray(Array):
+    def __init__(self, array, keys):
```

Comment:
The class name is a little interesting (and a docstring would help) - it looks like it is really needed to behave as a "NewAxesArray" as everything else already handles indexing without the ``np.newaxis`` (and in some cases Ellipsis).